.ci/test_r_package.sh

```diff
@@ -109,10 +124,35 @@ if [[ $TASK == "r-package-check-docs" ]]; then
 fi
 
 cd ${BUILD_DIRECTORY}
-Rscript build_r.R --skip-install || exit -1
 
 PKG_TARBALL="lightgbm_${LGB_VER}.tar.gz"
 LOG_FILE_NAME="lightgbm.Rcheck/00check.log"
+if [[ $R_BUILD_TYPE == "cmake" ]]; then
+    Rscript build_r.R --skip-install || exit -1
+elif [[ $R_BUILD_TYPE == "cran" ]]; then
+    ./build-cran-package.sh || exit -1
+
+    num_files_changed=$(
+        git diff --name-only | wc -l
+    )
+    if [[ ${num_files_changed} -gt 0 ]]; then
+        echo "'configure' in the R package has changed. Please recreate it and commit the changes."
+        echo "Changed files:"
+        git diff --compact-summary
+        echo "See R-package/README.md for details on how to recreate this script."
+        echo ""
+        exit -1
+    fi
```

Comment:
@jameslamb According to the 239ee3ff359462c385d4b4c7cc2aac04de82e3a7 it seems that we had `configure.ac` and `configure` not synchronized. Does it mean that this CI part doesn't work or I'm missing something?

Reply:
maybe `autoconf` has something in it that prevents creation of a new file if it's a comments-only change? That would be really weird and surprising to me.

Let me try adding an `echo "abc"` to `configure.ac` and see if it triggers the check in CI.

Reply:
hopefully this will trigger that error: https://github.com/microsoft/LightGBM/pull/3188/commits/856b80eb45fe9d5837f1da7938013acddeca4838

Reply:
it did not :upside_down_face: https://github.com/microsoft/LightGBM/pull/3188/checks?check_run_id=919834230

Reply:
OOOOOOOOOOOOO I know why, I removed re-creating `configure` from `build-cran-package.sh`. I wanted that script to be usable for anyone testing the CRAN package, and didn't want our users to need to have autoconf and automake installed (those should only be used by developers on this library).

I need to add re-running autoconf into `test_r_package.sh`.

Reply:
ok now it is failing successfully! https://github.com/microsoft/LightGBM/pull/3188/checks?check_run_id=919913867

![image](https://user-images.githubusercontent.com/7608904/88699342-d5f91c80-d0cc-11ea-995c-3ce3eb4af010.png)

Reply:
I just pushed another commit to revert the fake code I added to try to simulate this. We should now be good, as of https://github.com/microsoft/LightGBM/pull/3188/commits/c9bcc4628eb767167e331be58b08d51cd71c3dbe

Reply:
it is still breaking. Just pushed another commit so we can maybe see the diff. I'm really confused.

Reply:
ooooo everything breaking in CI is because `VERSION.txt` isn't read correctly, like

![image](https://user-images.githubusercontent.com/7608904/88721907-672bbb80-d0ec-11ea-90ec-c97134a1405f.png)

```
cat: src/VERSION.txt: No such file or directory
cat: src/VERSION.txt: No such file or directory
cat: src/VERSION.txt: No such file or directory
cat: src/VERSION.txt: No such file or directory
cat: src/VERSION.txt: No such file or directory
cat: src/VERSION.txt: No such file or directory
cat: src/VERSION.txt: No such file or directory
cat: src/VERSION.txt: No such file or directory
cat: src/VERSION.txt: No such file or directory
cat: src/VERSION.txt: No such file or directory
```

ok I can fix that

Reply:
let's see if this fixes it https://github.com/microsoft/LightGBM/pull/3188/commits/ff6da50cbda9dc482275d746716f806c39c9010e

Reply:
looks like it is fixed! https://github.com/microsoft/LightGBM/pull/3188/checks?check_run_id=920920582